web/libs/editor/src/lib/AudioUltra/Visual/Composition/LayerComposer.ts

```diff
@@ -0,0 +1,173 @@
+    const positions = new Map<Layer, { x: number; y: number }>();
+
+    for (const layer of composition.layers) {
+      const originalPos = composition.positions.get(layer);
+      if (!originalPos) continue;
+
+      positions.set(layer, {
+        x: originalPos.x + left,
+        y: originalPos.y + top,
+      });
+    }
+
+    // Calculate new total dimensions
+    const totalWidth = composition.totalWidth + left + right;
+    const totalHeight = composition.totalHeight + top + bottom;
+
+    return {
+      layers: composition.layers,
+      positions,
+      totalWidth,
+      totalHeight,
+    };
+  }
+
+  /**
+   * Register a layer with the composer
+   */
+  registerLayer(name: string, layer: Layer): LayerComposer {
+    this.layers.set(name, layer);
+    return this;
+  }
+
+  /**
+   * Register multiple layers at once
+   */
+  registerLayers(layers: Map<string, Layer>): LayerComposer {
+    for (const [name, layer] of layers.entries()) {
+      this.registerLayer(name, layer);
+    }
+    return this;
+  }
+
+  /**
+   * Draw layers on top of each other (overlapping at the same position)
+   * Each layer is positioned at x=0 and y=0
+   * @param items - Array of layer names, Layer objects, or CompositionResult objects
+   * @returns A CompositionResult
+   */
+  onTopOfEachOther(items: (string | Layer | CompositionResult)[]): CompositionResult {
+    const unsortedLayers: Layer[] = [];
+    const positions = new Map<Layer, { x: number; y: number }>();
+    let totalHeight = 0;
+    let totalWidth = 0;
+
+    for (const item of items) {
+      // Handle CompositionResult objects
+      if (item && typeof item === "object" && "layers" in item && "positions" in item) {
+        const composition = item as CompositionResult;
+
+        for (const layer of composition.layers) {
+          if (!layer || !layer.isVisible) continue;
+
+          unsortedLayers.push(layer);
+          // Preserve the original position from the composition
+          const originalPos = composition.positions.get(layer);
+          positions.set(layer, originalPos || { x: 0, y: 0 });
+
+          totalHeight = Math.max(totalHeight, layer.height);
+          totalWidth = Math.max(totalWidth, layer.width);
+        }
+        continue;
+      }
+
+      let layer: Layer | undefined;
+
+      if (typeof item === "string") {
+        // If an item is a string, get the layer by name
+        layer = this.layers.get(item);
+      } else {
+        // If item is already a Layer object
+        layer = item as Layer;
+      }
+
+      if (!layer || !layer.isVisible) continue;
+
+      unsortedLayers.push(layer);
+      positions.set(layer, { x: 0, y: 0 });
+
+      totalHeight = Math.max(totalHeight, layer.height);
+      totalWidth = Math.max(totalWidth, layer.width);
+    }
+
+    // Sort layers by z-index (index property) to ensure proper rendering order
+    const layers = [...unsortedLayers].sort((a, b) => a.index - b.index);
```

Comment:
There is no need to copy the array, the unsortedLayers does not need preservation and this is just overhead.
```suggestion
    const layers = unsortedLayers.sort((a, b) => a.index - b.index);
```